src/NetMQ/ReceiveThreadSafeSocketExtensions.cs

```diff
@@ -117,6 +119,31 @@ public static bool TryReceiveBytes(this IThreadSafeInSocket socket, [NotNullWhen
 
         #endregion
 
+        #region AsyncEnumerable
+
+#if NETSTANDARD2_1
+        
+        /// <summary>
+        /// Provides a consuming IAsyncEnumerable for receiving messages from the socket.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>An IAsyncEnumerable that receive and returns messages from the socket.</returns>
+        /// <exception cref="System.OperationCanceledException">The token has had cancellation requested.</exception>
+        public static async IAsyncEnumerable<byte[]> ReceiveBytesAsyncEnumerable(
+            [NotNull] this IThreadSafeInSocket socket,
```

Comment:
```suggestion
            this IThreadSafeInSocket socket,
```

In a nullable context (i.e. where `#nullable enable` is used) then `[NotNull]`/`[DisallowNull]` is the default for un-annotated references.

Note that if you want to annotate that null is not a valid input (eg. on an unconstrainted generic reference) you must use `DisallowNull` (or `MaybeNull` for the inverse). `NotNull` in this case implies the reference is not null _after_ the call returns, and is usually used for `ref`/`out` params.

Ditto three more times below.

Reply:
thanks, fixed.